### PR TITLE
Update team project access docs to include new project roles

### PR DESCRIPTION
**Pull request by joekarl**
### What
Update cloud-docs/api-docs/project-team-access to include additional project roles

### Why
New project roles exist in TFC/E and are exposed via UI/API, this adds docs for them.
Task to introduce API changes [TF-3984](https://hashicorp.atlassian.net/browse/TF-3984)
PR to introduce API changes [Atlas PR](https://github.com/hashicorp/atlas/pull/14496)
Task to introduce API docs [TF-3993](https://hashicorp.atlassian.net/browse/TF-3993)

----------

#### Pull Request
- [X] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [X] Description links to related pull requests or issues, if any.

#### Content
- [ ] ~Redirects have been added for moved, renamed, or deleted pages. This requires a separate PR in the [`terraform-website` repository](https://github.com/hashicorp/terraform-website) `redirects.next.js` file.~
- [X] API documentation and the API Changelog have been updated. 
- [ ] ~Links to related content where appropriate (e.g., API endpoints, permissions, etc.).~
- [ ] ~Pages with related content are updated and link to this content when appropriate.~
- [ ] ~Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.~
- [ ] ~New pages have metadata (page name and description) at the top.~
- [ ] ~New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.~
- [ ] ~New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.~
- [ ] ~UI elements (button names, page names, etc.) are bolded.~
- [ ] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

[TF-3984]: https://hashicorp.atlassian.net/browse/TF-3984?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TF-3993]: https://hashicorp.atlassian.net/browse/TF-3993?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/docs/cloud-docs/api-docs/project-team-access.mdx [hunk 1]
```diff
@@ -8,6 +8,14 @@ The team access APIs are used to associate a team to permissions on a project. A
...
 
+## Project Team Access Levels
+| Access Level    | Description                                                                                                                                                      |
+|-----------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `read`            | Read project and Read workspace access role on project workspaces                                                                                                |
+| `plan`  | Read project and Plan workspace access role on project workspaces                                                                                                |
+| `write` | Read project and Write workspace access role on project workspaces                                                                                               |
+| `admin`           | Admin project, Admin workspace access role on project workspaces, create workspaces within project, move workspaces between projects, manage project team access |
```
Comment (rberecka) [inline website/docs/cloud-docs/api-docs/project-team-access.mdx:17]: I'd suggest using parenthesis to make it clear that it's not admin access + these other things, but that admin access includes these things
```suggestion
| `admin`           | Admin project, Admin workspace access role on project workspaces (create workspaces within project, move workspaces between projects, manage project team access) |
```

**Review by rberecka — APPROVED** [review]
tiny nit, but approving!

File: website/docs/cloud-docs/api-docs/project-team-access.mdx (hunk 1 above)
Comment (joekarl) [reply website/docs/cloud-docs/api-docs/project-team-access.mdx:17]: Ah, I specifically used the "admin workspace access" + "specific project permissions" format to match the similar verbage for read/plan/write of "{read|plan|write} workspace access" + "specific project permissions" (there just happen to be none of the latter)

We can tweak this as we go

**Review by rberecka — APPROVED** [review]

File: website/docs/cloud-docs/api-docs/changelog.mdx [hunk 2]
```diff
@@ -7,6 +7,10 @@ page_id: api-changelog
 
 Keep track of changes to the API for Terraform Cloud and Terraform Enterprise.
 
+## 2023-03-13
+
+* Update [Project Team Access API](/terraform/cloud-docs/api-docs/project-team-access) to include additional Project roles
```
Comment (hs26gill) [inline website/docs/cloud-docs/api-docs/changelog.mdx:12]: Very nit-picky, feel free to ignore, looks like you can use full stop just to be consistent with most of the change logs.

**Review by hs26gill — APPROVED** [review]

File: website/docs/cloud-docs/users-teams-organizations/permissions.mdx [hunk 3]
```diff
@@ -158,14 +158,34 @@ Fixed permission sets are bundles of specific permissions for a project, designe
...
 
 - Read and update project settings.
 - Delete the project.
 - Create workspaces in the project.
 - Move workspaces into or out of the project. This also requires project admin permissions for the source or destination project.
 - Grant or revoke project permissions for visible teams. Project admins **cannot** view or manage access for teams that are are [Secret](/terraform/cloud-docs/users-teams-organizations/teams#team-visibility), unless those admins are also organization owners.
 
+#### Maintain
+
+The "maintain" permission is for people who need to manage existing infrastructure in a single project, while also granting the ability to create new workspaces in that project. Maintain access grants full control of everything in the project, including the following permissions:
```
[the remark this answers is not in the document]
Comment (jbonhag) [reply website/docs/cloud-docs/users-teams-organizations/permissions.mdx:171]: Good call out - we use "members", "users", and "people" pretty interchangeably on this page, but we can be consistent within this section.

**Pull request merged**